src/main.rs

```diff
@@ -1187,31 +1143,102 @@ fn parse_and_output_man_pages(
     unimplemented!();
 }
 
-fn parsers_to_try() -> Vec<Box<dyn ManParser>> {
-    vec![
-        Box::new(Type1),
-        Box::new(Type2),
-        Box::new(Type3),
-        Box::new(Type4),
-        Box::new(TypeDarwin),
-        Box::new(TypeDeroff),
-    ]
+macro_rules! mantypes {
+    ($($typ: tt),*) => {
+        $(
+        impl From<$typ> for ManType {
+            fn from(t: $typ) -> Self {
+                ManType::$typ(t)
+            }
+        }
+        )*
+
+        #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+        enum ManType {
```

Comment:
Tiny nit: Could the `enum ManType` be the first thing declared in the macro? Unless there's a reason for putting the `impl From ...` first?